ValidationApp/src/Scenes/LevelEditor.hpp

```diff
@@ -10,62 +12,323 @@
+                _tiles.back()->GetComponent<SnapToGridDraggable>().SetDraggeble(false);
+            }
+        }
+    }
 
-		std::map<std::string, SpriteData*>& sprites =
-			ResourceManager::GetAllSprites();
-		std::vector<std::shared_ptr<GameObject>> optionTiles;
 
-		int index = 0;
+    void OnStart() override
+    {
+        const auto viewport = GetWindow()->GetViewport();
+        const int windowWidth = viewport.width;
+        const int windowHeight = viewport.height;
 
-		for (auto& pair : sprites)
-		{
-			if (maxOptionPerRow <= index)
-				break;
+        const float topBarHeight = PADDING_TOP + SCALE_IN_UI_BAR + 10.0f;
+        const float titleLeftPadding = windowWidth / 2 - TITLE_WIDTH / 2;
 
-			optionTiles.emplace_back(
-				Instantiate({{scaleInUIBar * index + rightPadding * index +
-								  paddingOutLeftUI,
-							  paddingTop},
-							 0.0f,
-							 {scaleInUIBar, scaleInUIBar}}));
-			optionTiles.back()->AddComponent<Ui::Image>(pair.first.c_str());
-			optionTiles.back()->AddComponent<Ui::Button>()->SetOnLeftMouseClick(
-				[this]() -> void {
-					this->_tiles.emplace_back(
-						Instantiate({{1.0f, 1.0f}, 0.0f, {3.0f, 3.0f}}));
-				},
-				"UI");
-			index++;
-		}
-	}
+        const float rightBarStart = windowWidth - RIGHT_BAR_WIDTH;
 
-   private:
-	std::vector<std::shared_ptr<GameObject>> _tiles;
-};
\ No newline at end of file
+        InitLevelEditor(_saveFileName);
+
+        BackButton(rightBarStart);
+        SaveButton(rightBarStart);
+        LayerChanger(titleLeftPadding, topBarHeight, windowWidth, windowHeight);
+        TopBar(windowWidth, titleLeftPadding, topBarHeight);
+        WorldView();
+    }
+
+    void WorldView()
+    {
+        float speed = 0.3f;
+        _inputs.push_back(InputManager::keyPressed(KEY_D, [this, speed](Input& e){
+            _camera->AddToPosition({ speed,0.f});
+        }));
+        _inputs.push_back(InputManager::keyPressed(KEY_A, [this, speed](Input& e) {
+            _camera->AddToPosition({ -speed,0.f });
+        }));
+        _inputs.push_back(InputManager::keyPressed(KEY_S, [this, speed](Input& e) {
+            _camera->AddToPosition({ 0.f,-speed });
+        }));
+        _inputs.push_back(InputManager::keyPressed(KEY_W, [this, speed](Input& e) {
+            _camera->AddToPosition({ 0.f,speed });
+        }));
+    }
+
+    void BackButton(const float rightBarStart)
+    {
+        std::shared_ptr<GameObject> saveButton{
+            Instantiate({ { rightBarStart - 20.0f, PADDING_TOP }, 0.0f,{ 160.0f, SAVE_FONT_SIZE } })
+        };
+        saveButton->AddComponent<Ui::Text>("Back", "knight", SAVE_FONT_SIZE, Rendering::Color{ 0, 0, 0, 255 })->SetBackground({ 255, 255, 255, 255 });
+        saveButton->AddComponent<Ui::Button>()->SetOnLeftMouseClick([this]() {
+            SaveLevel();
+            }, "UI");
+    }
+
+    void SaveButton(const float rightBarStart)
+    {
+        std::shared_ptr<GameObject> saveButton{
```

Comment:
ik weet zo niet of de save button werkt
misschien handig om ook even een console log toe te voegen wanneer je opslaat

Reply:
ik zie nu namelijk alleen dat het werk aan de aangepaste json/wanneer ik de applicatie sluit en opnieuw run

Reply:
edit: het werkt niet geheel goed

Reply:
ik geef het weer in de console nu